### PR TITLE
fully walking

### PR DESCRIPTION
**Pull request by nathanjzhao**
walks 100%

### CODE REVIEW

File: environment.py [hunk 1]
```diff
@@ -21,21 +20,22 @@
...
+        "subtraction_factor": 0.05,
+        "max_diff_norm": 0.5,
+    },
+    "weights": {
+        "ctrl_cost": 0.1,
+        "original_pos_reward": 4,
+        "is_healthy": 1,
+        "velocity": 1.25,
+    },
+}
```
Comment (codekansas) [inline environment.py]: sad :(

Comment (nathanjzhao) [reply environment.py]: i gotchu

**Comment by nathanjzhao** [comment]
https://github.com/user-attachments/assets/10503ffa-1d16-4674-8698-915112a5aaf3

video

File: environment.py [hunk 2]
```diff
@@ -238,19 +238,37 @@ def compute_reward(
...
+
+        # MAINTAINING ORIGINAL POSITION REWARD
+        qpos0_diff = self.initial_qpos - state.qpos
+        original_pos_reward = jnp.exp(-exp_coef * jnp.linalg.norm(qpos0_diff)) - subtraction_factor * jnp.clip(
+            jnp.linalg.norm(qpos0_diff), 0, max_diff_norm
+        )
+
+        # HEALTHY REWARD
         is_healthy = jnp.where(state.q[2] < min_z, 0.0, 1.0)
         is_healthy = jnp.where(state.q[2] > max_z, 0.0, is_healthy)
```
Comment (budzianowski) [inline environment.py:253]: This one should be more about orientation and in exponent like 
```suggestion
        orientation = jnp.exp(-jnp.norm(self.projected_gravity[:, :2], dim=1) * weight)
```

File: infer.py [hunk 3]
```diff
@@ -1,85 +1,138 @@
...
+            episodes += 1
+            print("Episode", episodes, "reward:", episode_reward)
+            episode_reward = 0
 
         if len(rollout) >= max_frames:
             break
 
-    # Trim rollout to desired length
-    rollout = rollout[:max_frames]
+    # def scan_env(carry, t):
```
Comment (budzianowski) [inline infer.py:104]: Why commenting out?

**Review by budzianowski — APPROVED** [review]
rather 'fully running' ;)

File: environment.py (hunk 2 above)
Comment (nathanjzhao) [reply environment.py:253]: oooo cool

File: infer.py (hunk 3 above)
Comment (nathanjzhao) [reply infer.py:104]: i had scan env setup for vecotrized environments in infer -- should prob make scan env setup for normal environments in infer though you're right. hasn't been a problem so far b/c training is enough time that waiting a little longer to see the video isn't too bad

File: train.py [hunk 4]
```diff
@@ -316,22 +280,41 @@ def callback(info):
...
+        "CLIP_EPS": 0.2,
+        "ENT_COEF": 0.0,
+        "VF_COEF": 0.5,
+        "MAX_GRAD_NORM": 0.5,
+        "ACTIVATION": "tanh",
+        "ENV_MODULE": args.env_module,
+        "ANNEAL_LR": True,
+        "NORMALIZE_ENV": True,
+        "DEBUG": True,
+    }
```
Comment (codekansas) [inline train.py]: sad :(

**Pull request merged**